src/ynEIGEN/RedemptionAssetsVault.sol

```diff
@@ -190,33 +204,18 @@ contract RedemptionAssetsVault is IRedemptionAssetsVault, Initializable, AccessC
         emit TotalAssetsTransferred(ETH_ASSET, msg.sender, to, amountTransferred);
     }
 
-    /** 
-     * @notice Withdraws a specified amount of redemption assets and processes them through ynETH.
-     * @param amount The amount of ETH to withdraw and process.
-     * @dev Requires the caller to be the redeemer and the contract to not be paused.
-     */
-    function withdrawRedemptionAssets(uint256 amount) public onlyRedeemer whenNotPaused nonReentrant {
-        IERC20[] memory assets = assetRegistry.getAssets();
-        for (uint256 i = 0; i < assets.length; ++i) {
-            IERC20 asset = assets[i];
-            uint256 assetBalance = balances[address(asset)];
-            if (assetBalance > 0) {
-                uint256 assetBalanceInUnit = assetRegistry.convertToUnitOfAccount(asset, assetBalance);
-                if (assetBalanceInUnit >= amount) {
-                    uint256 reqAmountInAsset = assetRegistry.convertFromUnitOfAccount(asset, amount);
-                    ynEigen.processWithdrawn(reqAmountInAsset, address(asset));
-                    balances[address(asset)] -= reqAmountInAsset;
-                    emit AssetWithdrawn(address(asset), msg.sender, address(ynEigen), reqAmountInAsset);
-                    break;
-                } else {
-                    ynEigen.processWithdrawn(assetBalance, address(asset));
```

Comment:
the bug is here:

This function calls ynEigen to add extra tokens to its holdings but does not approve the tokens to be transferred over